extensions/pkg/controller/worker/machines.go

```diff
@@ -165,9 +170,24 @@ func WorkerPoolHashV1(pool extensionsv1alpha1.WorkerPool, cluster *extensionscon
 }
 
 // WorkerPoolHashV2 returns a hash value for a given nodeAgentSecretName and additional data.
-func WorkerPoolHashV2(nodeAgentSecretName string, additionalData ...string) (string, error) {
+func WorkerPoolHashV2(nodeAgentSecretName string, pool extensionsv1alpha1.WorkerPool, cluster *extensionscontroller.Cluster, additionalData ...string) (string, error) {
 	data := []string{nodeAgentSecretName}
 
+	if v1beta1helper.IsUpdateStrategyInPlace(pool.UpdateStrategy) {
```

Comment:
please add a small comment here that why this special handling for the in-place update strategy